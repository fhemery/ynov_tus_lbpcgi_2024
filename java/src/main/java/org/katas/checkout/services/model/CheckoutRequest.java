package org.katas.checkout.services.model;

public record CheckoutRequest(String cartId, String deliveryName, String deliveryStreet, String deliveryCity, String creditCardNumber, String creditCardExpiry, String creditCardCvv) {
}
