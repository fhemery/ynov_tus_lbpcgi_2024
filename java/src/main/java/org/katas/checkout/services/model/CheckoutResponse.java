package org.katas.checkout.services.model;

public record CheckoutResponse(String orderId, String orderNumber, int totalAmount) {
}
