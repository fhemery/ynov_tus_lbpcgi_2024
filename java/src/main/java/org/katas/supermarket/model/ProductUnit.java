package org.katas.supermarket.model;

public enum ProductUnit {
    KILO, EACH
}
