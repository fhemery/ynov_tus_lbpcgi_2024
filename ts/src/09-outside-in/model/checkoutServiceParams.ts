export class CheckoutServiceParams {
    constructor(readonly basketId: string) {
        if(!basketId){
            throw new Error('basketId is required');
        }
    }
}
