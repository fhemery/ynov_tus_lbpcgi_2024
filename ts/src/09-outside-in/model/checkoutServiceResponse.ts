export class CheckoutServiceResponse {
    constructor(readonly orderId: string, readonly total: number) {
    }
}
