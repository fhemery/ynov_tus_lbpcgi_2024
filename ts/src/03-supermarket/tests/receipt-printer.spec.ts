import {ReceiptPrinter} from "../src/ReceiptPrinter";
import {Receipt} from "../src/model/Receipt";
import {Product} from "../src/model/Product";
import {ProductUnit} from "../src/model/ProductUnit";

describe('Receipt Printer', () => {
    let receiptPrinter: ReceiptPrinter;

    beforeEach(() => {
        receiptPrinter = new ReceiptPrinter();
    });

    it('should return empty ticket with a receipt with no product', () => {
        const receipt = new Receipt();

        const printedReceipt = receiptPrinter.printReceipt(receipt);

        expect(printedReceipt).toMatchSnapshot();
    });

    it('should return a line with an added product', () => {
        const receipt = new Receipt();
        receipt.addProduct(new Product('Apples', ProductUnit.Kilo), 3, 2.50, 7.50);

        const printedReceipt = receiptPrinter.printReceipt(receipt);
        expect(printedReceipt).toMatchSnapshot();
    });
});
