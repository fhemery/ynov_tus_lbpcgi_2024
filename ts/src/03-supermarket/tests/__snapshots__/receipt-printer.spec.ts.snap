// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Receipt Printer should return a line with an added product 1`] = `
"Apples                              7.50
  2.50 * 3.000

Total:                              7.50"
`;

exports[`Receipt Printer should return empty ticket with a receipt with no product 1`] = `
"
Total:                              0.00"
`;

exports[`Receipt Printer should return empty ticket with empty receipt 1`] = `
"
Total:                              0.00"
`;
